src/web-server.js

```diff
@@ -83,21 +84,29 @@ app.patch('/services', function(req, res) {
   if ( req.body != undefined && req.body.hasOwnProperty('service') ) {
     service = req.body.service
   } else {
-    res.status(400).send('service key is missing')
+    return res.status(400).send('service key is missing')
   }
   if ( req.body != undefined && req.body.hasOwnProperty('url') ) {
     url = req.body.url
   } else {
-    res.status(400).send('url key is missing')
+    return res.status(400).send('url key is missing')
```

Comment:
I think we want to return from these, before I think `modifyService` would still get called even though we replied with a 400 :-1: